docs/gt_frontend/language_reference.rst

```diff
@@ -0,0 +1,332 @@
+==========================================
+Language Reference
+==========================================
+
+Abstract
+========
+
+This document contains a draft of the DSL syntax for GTScript on meshes.
+
+**Design Goals**
+
+1. Concise, compact and readable syntax
+2. General enough to allow generation of efficient code for meshes with unstructured, partially structured, structured meshes
+3. Easy to translate into GTScriptAST
+
+Motivation and Scope
+====================
+
+For global weather models (e.g. `IFS-FVM <https://refubium.fu-berlin.de/bitstream/handle/fub188/25122/K%C3%BChnlein_FVM_2019.pdf?sequence=1>`__, ICON) the ability to handle non-regular grids is required.
+
+Concepts
+========
+
+**Location**
+
+A bounded set with nonempty interior. We differentiate between horizontal locations, i.e. :code:`Cell`, :code:`Edge`, :code:`Vertex` and vertical locations, i.e. a layer. Each location can be attributed to a domain dimesion.
+
+**Mesh**
+
+A subdivision of the computational domain into locations.
+
+**Field**
+
+A mapping between locations and quantities at this location.
+
+.. math::
+  f: (c_1, ..., c_n) \mapsto a
+
+**Local field**
+
+A field defined in the neighbourhood of a cell.
+
+**Neighbourhood chain**
+
+Used to select direct or indirect neighbours of a cell. See here https://docs.google.com/document/d/1nQZzKGi7Go9R3fme78ViB_PrQDVPxy4xTzSuGh_yJAI/edit for a description.
```

Comment:
I would add a TODO here to fully define the Neighborhood chain concepts inside the concepts repo (with pictures if needed). It's good to have the link to the google doc here, but I think it shouldn't be required to follow the link to understand the idea.

Reply:
Totally agreed.